sstables/sstables.cc

```diff
@@ -2138,10 +2138,27 @@ sstable::make_reader_v1(
         streamed_mutation::forwarding fwd,
```

Comment:
`make_reader` and `make_reader_v1` now have a non-uniform API: `make_reader_v1` provides reversing but `make_reader` does not.

I think this should at least be documented. And it should be noted that `as_mutation_source` uses `make_reader_v1` and it should stay that way for now (perhaps a comment inside `as_mutation_source`) due to `make_reader`'s current behavior (no reversing).

Reply:
Also offtop rant: why the heck do we use `make_reader`+`make_reader_v1` naming convention instead of `make_reader_v2`+`make_reader` which is used in `mutation_source`? I love these subtle differences sprinkled all over Scylla codebase, definitely makes working with it a pleasure.

Reply:
Yes the difference is unfortunate and I don't like it. Adding reversing support to v2 would require implementing a v2 reversing reader, which seems futile effort, given that the v2 reader will support it natively. Our mid-term plan is to get rid of the reversing reader, only keeping it as a kl/la reader local adaptor. This will be a v1 reversing reader (the current one) and I don't see much point in adding reverse support to an unused code-path (we still use v1 path everywhere).

Reply:
> Also offtop rant: why the heck do we use `make_reader`+`make_reader_v1` naming convention instead of `make_reader_v2`+`make_reader` which is used in `mutation_source`?

This is because the mx sstable reader was already converted to be v2 native. Since mx is our main format, `make_reader()` returns a v2 reader -- the sstable's native reader.
Our codebase however still uses v1 as the default -- this is what `mutation_source::make_reader()` expresses.